src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -290,11 +290,13 @@ static Value builtinStringRepeat(ExecutionState& state, Value thisValue, size_t
     Value argument = argv[0];
     int32_t repeatCount;
     double count = argument.toInteger(state);
-    if (count < 0 || count == std::numeric_limits<double>::infinity()) {
+    double newStringLength = str->length() * count;
+    if (count < 0 || count == std::numeric_limits<double>::infinity() || newStringLength > STRING_MAXIMUM_LENGTH) {
         ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "invalid count number of String repeat method");
     }
     repeatCount = static_cast<int32_t>(count);
-    if (count == 0) {
+
+    if (count == 0 || str->length() == 0) {
```

Comment:
How about replace this if statement by checking the `newStringLength` such as `newStringLength == 0`?

Reply:
Indeed, good idea!